infra/src/test/kotlin/fi/fta/geoviite/infra/tracklayout/TrackLayoutDomainTestData.kt

```diff
@@ -889,13 +889,14 @@ fun switchFromDbStructure(
     },
 )
 
+private var switchNameCounter = 0
+
 fun switch(
-    seed: Int = 1,
     structureId: IntId<SwitchStructure> = switchStructureYV60_300_1_9().id as IntId,
-    joints: List<TrackLayoutSwitchJoint> = joints(seed),
-    name: String = "TV$seed",
+    joints: List<TrackLayoutSwitchJoint> = listOf(),
+    name: String = "TV${switchNameCounter++}",
```

Comment:
Vaihteiden nimet luodaan defaulttina nyt samalla tavalla kuin esim. sijaintiraiteiden. Jätin myös `TestDBService`:n `getUnusedSwitchName()`-funktion olemaan, sillä en katsonut sen tekevän mitään pahaa. Se vaan käy kannassa toisin kuin tämä, joten siitä ei oikein saanut leivottua defaulttia.

Samalla defaultataan nykyään vaihdepisteet tyhjäksi listaksi. Tämä toimi lopulta yllättävän hyvin suurelle osalle testejä

Reply:
Tässä on siis edelleen satunnaisuutta sen perusteella missä järjestyksestä testit ajetaan ja teoriassa kantaan voi päätyä samannimisiä vaihteita jos testin/testit ajaa useammin kuin yhden kerran?🤔 Ei tuo varmaan massiivinen pulma liene, mutta hieman epädeterminististä edelleen kun siitä pyrittiin mahdollisimman paljon eroon

Vaihdepisteiden listan asettaminen tyhjäksi on myös vähän hähmeää minusta. Jotenkin olettaisin että switch()-tuottaisi perusvaihderakenteen, eli sillä olisi myös jotain pisteitä olemassa. Tosin nähtävästi vaihde on meillä sallittu ilman pisteitä. Vaikuttaa kuitenkin hieman hassulta.

Toki tuo joints()-funktio oli myös olemassa, eli aiemmin siellä on ollut oletettavasti aina epätyhjä lista -> meneeköhän osa testeistä nyt läpi siksi, ettei yhtään vaihdepistettä enää ole, jolloin esim. for-looppeja ei käytäisi enää kertaakaan läpi yhden tai useamman testin aikana? (=> mahdollisia vääriä positiivisia testituloksia)

Reply:
Minusta on tavallaan ihan hyväkin että vaihteelle ei tule pisteitä. Kuitenkin niillä on oikeasti merkitystä lähinnä jos se kytketään raiteeseen ja normi flow:ssa ne syntyy linkityksessä. Testissä tuo vaatii muutakin setuppia -> pakko antaa eksplisiittisesti jos niitä halutaan. Tuo on toki ihan pointti että joku nykytesti saattaa olettaa jointteja ja mennä läpi vain koska mitään ei nyt tehdä. Vähän hankala saada tuollaista kiinni :thinking: 

Tuo nimijuttu toiminee periaatteessa tässä. Tavallaan minusta oli kyllä ok sekin että tämä antaa aina vakio-nimen - jos sen meinaa laittaa kantaan (jolloin sen tarvitsee olla uniikki) niin se kutsuja voi pyytää uniikin nimen olemassaolevalla util-funkkarilla yhdellä kutsulla. Sen tarjoama nimi on myös parempi koska sisältää testiluokan nimen, mikä on ollut välillä varsin hyödyllistä debugatessa (löytyy oikea rivi taulusta helpommin). Täällä tuota utilia ei tietenkään voi kutsu koska nämä funkkarit ei voi käyttää kantaa, mutta IT/UI testeissä tämän käyttäjä voi sen tehdä -> parempi konventio. Unit-testeissä ei puolestaan ole mitään tarvetta uniikkiudelle koska kanta-constrainteihin ei muutenkaan voi törmätä. Toisaalta tämäkin lienee ok ratkaisu jos vähentää tässä kohtaa tehtävän rempan määrää.

Reply:
Olen kanssa Jyrkin kanssa samaa mieltä tuosta vaihdepisteasiasta. Melko suurta osaa testimassasta ei pitäisi kiinnostaa ollenkaan onko luodulla vaihteita pisteitä vai ei, ja ne testit joita kiinnostaa pitäisi, pitäisi tarkistaa niiden olemassaolo (jos ne eivät tarkista niin ne testit itsessään ovat sitten viallisia.) Tuollaisia looppauksia ei mielestäni ole kauheasti harrastettu testeissä, mutta se ei toki kerro etteikö sellaisia olisi (toki monesti listojen tarkastuksen yhteydessä meillä on myös tarkastettu että listan koko on sopiva.) Oikein muita keinoja sellaisten etsimiseen ei nähdäkseni oikein ole kuin joko testien manuaalinen läpikäynti tai se puhuttu randomisaatio.

Tuo nimijuttu ei tosiaan ole tällä hetkellä ideaalinen, mutta se on konsistentti esim. sijaintiraiteiden jne. kanssa. Mielestäni oikein mahdollinen ratkaisu tähän olisi tosiaan jonkinlainen vakionimi defaultiksi, mutta sillä on potentiaalia olla loputtomampi suo: Integraatiotesteissä luodut tavarat tallennetaan kantaan ja todella monessa testissä mitään nimiä ei kiinnitetä. Noiden yhdistelmänä collisionit kannassa ovat todella mahdollisia mikäli jokin testi ei siivoilekaan kantaa ennen ajoaan. Mielestäni kun tuohon maailmaan mennään, niin sijaintiraiteet ja muutkin vastaavat counter-pohjaiset härdellit kannattaisi muuttaa samalla inertialla vakiodefaulttimalliin. Halusin vaan rajata sen ulos tästä nimenomaisesta jutusta ulos koska tämä itsessään oli jo aika iso tangentti tiketin alkuperäisestä jutusta enkä halunnut enää lähteä tangentin tangentille.